### ISSUE TITLE
Two questions about simulator training details and realistic data preprocessing

### ISSUE THREAD

**Issue by hnuzhy** [nickgkan/3d_diffuser_actor#38]
Hi, dear authors, thanks a lot for releasing your excellent work. Now, I mainly have two questions needing your help:

1. When training on simulated scenes, both CALVIN and RLBench, I found that the used GPU memory are dynamicly changing in a large range. Is this normal? I used one single V100 with 80GB, and adjusted the total iters correspondingly. Will this have a big influence on the  final results?
2. In real robot experiments, I should firstly prepare and preprocess the test data into the format used during training. For the 3D point clouds inputs (in format `[batch_size, num_cameras, 3, H, W]`), they are already well-prepared by simulator `PyRep`. My question is how can we obtain the same 3D point clouds as in the benchmark `RLBench`? As far as I konw, the captured 3D point clouds by hardwares are all dense ones with million points. And the used 3D point clouds in each frame have `H*W` (e.g., 256*256=65536) points which are relatively sparse. They are quite differenet. Actually, I found the scirpts in [pyrep - vision_sensor](https://github.com/stepjam/PyRep/blob/master/pyrep/objects/vision_sensor.py#L145) for lifting 2D RGB-D input into 3D point clouds. Can I use these scirpts for data preprocessing? Or you have any other ways when operating this step?

I'm looking forward for your reply. Best regrads~

**Comment by nickgkan** [comment]
Hi, thanks for your interest in our work.

1a. Dynamically changing GPU memory: yes this is normal, because not all episodes have the same length and we concatenate them (no padding).
1b. Influence of hyperparameters: lowering batch size could affect the iterations needed for convergence. On CALVIN this may have significant influence on the results, but we cannot be sure for all possible hyperparameter values.

2. We convert depth maps to point clouds using the camera parameters. This is the function we use on CALVIN for example https://github.com/nickgkan/3d_diffuser_actor/blob/master/utils/utils_with_calvin.py#L208.

I hope this helps!

**Comment by hnuzhy** [comment]
> Hi, thanks for your interest in our work.
> 
> 1a. Dynamically changing GPU memory: yes this is normal, because not all episodes have the same length and we concatenate them (no padding). 1b. Influence of hyperparameters: lowering batch size could affect the iterations needed for convergence. On CALVIN this may have significant influence on the results, but we cannot be sure for all possible hyperparameter values.
> 
> 2. We convert depth maps to point clouds using the camera parameters. This is the function we use on CALVIN for example https://github.com/nickgkan/3d_diffuser_actor/blob/master/utils/utils_with_calvin.py#L208.
> 
> I hope this helps!

OK. Thank you very much for your helpful reply. I will try the answer 2 in my real robot tests.
